src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java

```diff
@@ -197,4 +197,7 @@ protected void doRun() throws Exception {
     }
   }
 
+  public static boolean hasCredentials() {
+    return !StringUtils.isEmpty(GlobalPluginConfiguration.get().getCredentialsId());
```

Comment:
OK, @scoheb reminded me of this PR yesterday.  Went back and revisited things, and have been looking at what Ryan had put in for credentials some more and understand it better.

Rather than all the complexity I outlined in https://github.com/jenkinsci/openshift-sync-plugin/pull/40#issuecomment-298071143, I think the `hasCredentials()` method should be changed to 

```
return !StringUtils.isEmpty(getAuthenticatedOpenShiftClient().getConfiguration().getOauthToken());
```

That should cover the 3 scenarios that @bparees outlined in https://github.com/jenkinsci/openshift-sync-plugin/pull/40#issuecomment-298057268 but defer on the details of *which* of the 3 paths determined the credential.

Aside from switching to that, if you could rebase @scoheb and move `hashCredentials()` to CredentialsUtils.java, I think we should be good to go.

thanks